app/scripts/models/auth_brokers/redirect.js

```diff
@@ -76,7 +76,7 @@ define([
       // event handlers before the flow completes.
       var self = this;
       return p().delay(100).then(function () {
-        if (self.isOriginalTab()) {
+        if (self.isOriginalTab() || account.get('sessionToken') && alwaysRedirect) {
```

Comment:
Any better way to do this, is there anything else we can use instead of `sessionToken` to signal that this is SAME_BROWSER?

<img src=http://media.giphy.com/media/Q1LPV0vs7oKqc/giphy.gif width=200 />